platform/qt/app/mapwindow.cpp

```diff
@@ -229,7 +229,21 @@ void MapWindow::keyPressEvent(QKeyEvent *ev)
 
             m_map->setFilter("3d-buildings", buildingsFilterExpression);
 
-            m_map->setPaintProperty("3d-buildings", "fill-extrusion-color", "#aaa");
+            QString fillExtrusionColorJSON = R"JSON(
+              [
+                "interpolate",
+                ["linear"],
+                ["get", "height"],
+                  0.0, "blue",
+                 20.0, "royalblue",
+                 40.0, "cyan",
+                 60.0, "lime",
+                 80.0, "yellow",
+                100.0, "red"
+              ]
+            )JSON";
+
+            m_map->setPaintProperty("3d-buildings", "fill-extrusion-color", fillExtrusionColorJSON);
```

Comment:
do we always pass JSON strings? is it possible to set property, just as plain string value? (e.g. below I see setting a double :) )

Reply:
> do we always pass JSON strings? 

No - `set{Paint,Layout}Property` accepts a `QVariant`. In this case, the `QVariant` will be treated as a string if its inner type is either a `QString` or a `QByteArray`. In the double example, that's not a string - but a `double` which gets converted to a `QVariant`.